hcipy/field/util.py

```diff
@@ -7,540 +7,540 @@
-
-	.. math:: \Delta x = \lambda f / D = \lambda F
-
-	where :math:`\lambda` is the wavelength, :math:`f` is the effective focal length before the focal plane,
-	:math:`D` is the diameter of the pupil, and :math:`F` is the F-number of the incoming light beam.
-	You can supply either the spatial resolution or a set of focal length, reference wavelength and pupil diameter,
-	or a set of F-number and reference wavelength. If none are supplied, a spatial resolution of 1 will be
-	assumed, meaning normalized units.
-
-	The grid will always contain the origin (0, 0) point.
-
-	Parameters
-	----------
-	q : scalar or array_like
-		The number of pixels per resolution element (= lambda f / D).
-	num_airy : scalar or array_like
-		The spatial extent of the grid in radius in resolution elements (= lambda f / D).
-	spatial_resolution : scalar	or array_like
-		The physical size of a resolution element (= lambda f / D). It this is not given,
-		the spatial resolution will be calculated from the given `focal_length`,
-		`reference_wavelength` and `pupil_diameter`.
-	pupil_diameter : scalar or array_like
-		The diameter of the pupil. If it is an array, this indicates the diameter in x and y.
-	focal_length : scalar
-		The focal length used for calculating the spatial resolution at the focal plane.
-	f_number : scalar or array_like
-		The F number, also known as focal ratio, at the focal plane. If this is given, it overrides
-		the given pupil diameter and focal length.
-	reference_wavelength : scalar
-		The reference wavelength used for calculating the spatial resolution at the focal plane.
-
-	Returns
-	-------
-	Grid
-		A Grid describing the sampling for a focal plane.
-
-	Raises
-	------
-	ValueError
-		If both no spatial resolution and no complete set of (focal length, reference wavelength and pupil diameter) was supplied.
-	'''
-	if isinstance(q, Grid):
-		raise ValueError('The function signature was changed as of HCIPy 0.3.0. Please use the new signature (prefered), or use make_focal_grid_from_pupil_grid() if you want to retain old behaviour.')
-
-	if spatial_resolution is None:
-		if f_number is None:
-			if pupil_diameter is None or focal_length is None:
-				if reference_wavelength is None:
-					spatial_resolution = 1
-				else:
-					raise ValueError('You only supplied a reference wavelength and forgot to supply either an f_number or a (pupil_diameter, focal_length).')
-			else:
-				f_number = focal_length / pupil_diameter
-
-		if spatial_resolution is None:
-			if reference_wavelength is None:
-				raise ValueError('You supplied an f_number or (pupil_diameter, focal_length), and forgot to supply a reference wavelength.')
-			else:
-				spatial_resolution = f_number * reference_wavelength
-
-	delta = spatial_resolution / q * np.ones(2)
-	dims = (2 * num_airy * q * np.ones(2)).astype('int')
-	zero = delta * (-dims / 2 + np.mod(dims, 2) * 0.5)
-
-	return CartesianGrid(RegularCoords(delta, dims, zero))
+    r'''Make a grid for a focal plane.
```

Comment:
Is this `r` here in purpose?

Reply:
The `r` is for raw string, which escapes some of the math in this docstring.